### PR TITLE
Closes out Issue #9 - Add a "settings" modal

### LINKED ISSUE #9: #8: Add a “settings” modal
AC:

1. Add an icon + link to the top/right of the app in the `Header` (you can use `public/img/settings.png` for the icon)
2. When the user taps the settings icon, a modal should show over the top of the app
3. Along with a modal should appear an icon + link in the same spot as the settings icon that can be used to close the modal (you can use `public/img/close-x.png` for the icon)
4. The modal should have a heading that says “Settings” (we’ll add the rest of the settings UI next week!)

### PR DESCRIPTION
**Pull request by mjhcodes**
## Description

The purpose of this PR is to implement the on and off toggling of the "settings" modal, as well as the creation of its basic UI. Further additions to the modal will be addressed in a separate issue.

## Actions

- Added a new boolean state (displayModal) to the Header component to represent the on and off display of the modal
- Added a "gear" icon button to the header + updated styling
- Hooked up the logic to display the modal to the "gear" icon button
- Created a SettingsModal component with corresponding styling - inherits the setDisplayModal function
- Added a "close" icon button inside the SettingsModal component
- Hooked up the logic to close the modal to the "close" icon button
- Utilized a logical && operator to determine whether or not to display the modal, based on the displayModal state

## Related Issue

Closes #9 

## Acceptance Criteria

- [x]  Add an icon + link to the top/right of the app in the Header (you can use public/img/settings.png for the icon)
- [x] When the user taps the settings icon, a modal should show over the top of the app
- [x] Along with a modal should appear an icon + link in the same spot as the settings icon that can be used to close the modal (you can use public/img/close-x.png for the icon)
- [x] The modal should have a heading that says “Settings” (we’ll add the rest of the settings UI next week!)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Screen Shot 2020-06-02 at 1 41 17 PM

#### EXAMPLE 1 - WITH SETTINGS ICON ADDED TO HEADER
Screen Shot 2020-06-02 at 3 37 18 PM

#### EXAMPLE 2 - WITH SETTINGS MODAL DISPLAYED
Screen Shot 2020-06-03 at 6 00 24 PM

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -10,6 +10,26 @@
...
+
+.header-settings-btn {
+  right: 20px;
+}
+
+.header img {
+  height: 20px;
+  width: 20px;
+}
+
```
Comment (AshlingH) [inline src/App.css]: Really nice simple CSS, well done

File: src/components/Header.js [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+        What's near me?
+        <button
+          className="header-settings-btn"
+          onClick={() => setDisplayModal(true)}
+        >
+          <img src="/img/settings.png" alt="gear icon" />
+        </button>
+      </header>
+    </Fragment>
+  );
```
Comment (AshlingH) [inline src/components/Header.js:34]: excellent, this is how we worked with Modals with the shopping list task very simple and to the point

**Review by AshlingH — APPROVED** [review]
Great Job folks, was Materialize utilized in this??

File: src/App.css [hunk 3]
```diff
@@ -70,6 +90,51 @@
...
 }
 
+/* sets appearance and position of background when modal is open */
+.modal-container {
+  position: fixed;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
```
Comment (segdeha) [inline src/App.css:158]: Another way to do this that's not sensitive to if you add padding to the element is to specify:

    bottom: 0;
    left: 0;
    right: 0;
    top: 0;

Same effect, but no chance of accidental scroll bars.

File: src/App.css (hunk 1 above)
Comment (segdeha) [reply src/App.css]: Agreed!

File: src/components/Header.js [hunk 4]
```diff
@@ -1,7 +1,24 @@
...
+import React, { Fragment, useState } from "react";
+import SettingsModal from "./SettingsModal";
 
 const Header = () => {
-  return <header className="header">What's near me?</header>;
+  const [displayModal, setDisplayModal] = useState(false);
+
+  return (
+    <Fragment>
+      {displayModal && <SettingsModal setDisplayModal={setDisplayModal} />}
```
Comment (segdeha) [inline src/components/Header.js:11]: Nice! Both concise and clear. Love it!

**Review by segdeha — CHANGES_REQUESTED** [review]
The code is looking really good and the modal opens and closes as expected. Nice work!

It looks like the branch has conflicts, so you'll need to resolve those. Additionally, on Safari, the gear icon isn't cooperating (see screenshot).

image

File: src/App.css [hunk 5]
```diff
@@ -10,6 +10,26 @@
...
+  align-items: center;
+  justify-content: center;
+  padding: 2px;
+  border: none;
+  background: transparent;
+  cursor: pointer;
+}
+
+.header-settings-btn {
+  right: 20px;
```
Comment (segdeha) [inline src/App.css:31]: Adding `top: 6px;` to this ruleset fixes it when I apply that locally. Might not be an ideal fix, but maybe good enough?

Also, I'm happy to go ahead and approve this and we can fix the button placement later!

**Review by segdeha — APPROVED** [review]
Happy to fix the button placement later, so approving!

Need to fix the merge conflicts, but otherwise good to go 💪

**Pull request merged**